client/src/components/ForgotPassword.js

```diff
@@ -0,0 +1,89 @@
+import React, { PropTypes } from 'react';
+import { withRouter, Link } from 'react-router';
+import { compose } from 'redux';
+import { Field, reduxForm } from 'redux-form';
+
+class ForgotPassword extends React.Component {
+  static propTypes = {
+    handleSubmit: PropTypes.func.isRequired,
+    router: PropTypes.shape({
+      push: PropTypes.func.isRequired
+    }).isRequired,
+    location: PropTypes.shape({
+      state: PropTypes.shape({
+        nextPathname: PropTypes.string
+      })
+    }).isRequired,
+    isLoggedIn: PropTypes.bool.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      resetEmailSent: false
+    };
+    this.handleSendEmail = this.handleSendEmail.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isLoggedIn) {
+      const locationState = nextProps.location.state;
+
+      if (locationState && locationState.nextPathname) {
+        this.props.router.push(locationState.nextPathname);
+      } else {
+        this.props.router.push('/events');
+      }
+    }
+  }
+
+  handleSendEmail(event) {
+    event.preventDefault();
+    this.setState({
+      resetEmailSent: true
```

Comment:
At some point, it would be nice to change this, so we show a progress indicator immediately after the user submits the form, and then after a successful response we could show this message. This should be fine for now though.